jsoncppConfig.cmake.in

```diff
@@ -0,0 +1,18 @@
+cmake_policy(PUSH)
+cmake_policy(VERSION 3.0)
+
+@PACKAGE_INIT@
+
+include ( "${CMAKE_CURRENT_LIST_DIR}/jsoncpp-targets.cmake" )
+
+if(TARGET jsoncpp_static)
+    add_library(JsonCpp::JsonCpp INTERFACE IMPORTED )
+    set_target_properties(JsonCpp::JsonCpp PROPERTIES INTERFACE_LINK_LIBRARIES "jsoncpp_static")
+elseif(TARGET jsoncpp_lib)
+    add_library(JsonCpp::JsonCpp INTERFACE IMPORTED )
+    set_target_properties(JsonCpp::JsonCpp PROPERTIES INTERFACE_LINK_LIBRARIES "jsoncpp_lib")
+endif()
```

Comment:
I would put this logic in a file `jsoncpp-namespaced-targets.cmake` exported with `install(EXPORT ...)`.
Moreover, for consistency with `add_subdirectory()`, `JsonCpp::JsonCpp` ALIAS target should be added in CMakeLists.

Reply:
> I would put this logic in a file jsoncpp-namespaced-targets.cmake exported with install(EXPORT ...).

You mean such that lines 6-7 above read like this?

```
include ( "${CMAKE_CURRENT_LIST_DIR}/jsoncpp-targets.cmake" )
include ( "${CMAKE_CURRENT_LIST_DIR}/jsoncpp-namespaced-targets.cmake" )
```

Reply:
That's it 😃

Reply:
Right. And about the ALIAS target, where would you do that?

Reply:
In `/src/lib_json/CMakeLists.txt` close to `add_library()` calls. Unfortunately, it will probably requires some ugly conditions because currently `jsoncpp` can build both shared & static, or shared only, or static only.

Reply:
I would be glad to add ALIAS to CMakeLists to use the namespaced target in build tree also. 
 What is the usecase of having both static and shared build at same time?  CMake allows to specify the library type during generation with -DBUILD_SHARED_LIBS=Off/On although it relies on CMakeLists.txt  to not specify types explicitly.

Reply:
@Spacelm: Could we imagine fixing the `include` part of your comment in this PR, and leaving the ALIAS part for another PR? So that there could be a discussion about @sergeyrachev's point with which I agree: why not relying on `BUILD_SHARED_LIBS` instead? That makes it simpler IMO, and if you want both, you can run the build twice :thinking:.

Reply:
No problem for me to add ALIAS target in another PR, it would be an improvement.

Then, you can introduce a new breaking change in jsoncpp by removing again the ability to build both static & shared (AFAIK, it was removed for 1.9.0, then introduced again in 1.9.4...).